File: Vector/OOPs/Day 2 Applying SOLID Principles, DRY, and YAGNI in Java-JS/BASIC/Part 2 Open_close principle (OCP)/Employee.js
import BonusCalculator from './BonusCalculator.js';
class Employee {
    constructor(name, salary, bonusCalculator) {
        this.name = name;
        this.salary = salary;
        this.bonusCalculator = bonusCalculator;
    }

    getName() {
        return this.name;
    }

    getSalary() {
        return this.salary;
    }

    calculateBonus() {
        return this.bonusCalculator.calculateBonus(this.salary);
    }
}

export default Employee;

File: Vector/OOPs/Day 2 Applying SOLID Principles, DRY, and YAGNI in Java-JS/BASIC/Part 2 Open_close principle (OCP)/Employee.java
public class Employee {
    private String name;
    private double salary;
    private BonusCalculator bonusCalculator;

    public Employee(String name, double salary, BonusCalculator bonusCalculator) {
        this.name = name;
        this.salary = salary;
        this.bonusCalculator = bonusCalculator;
    }

    public String getName() {
        return name;
    }

    public double getSalary() {
        return salary;
    }

    public double calculateBonus() {
        return bonusCalculator.calculateBonus(salary);
    }
}